src/chain.cpp

```diff
@@ -71,7 +71,7 @@ CBlockIndex::CBlockIndex(const CBlock& block):
         nNonce{block.nNonce}
 {
     ClearMapZcSupply();
-    if(block.nVersion > 3 && block.nVersion < 7)
+    if (block.nVersion == 4)
```

Comment:
For all these instances why not do a || statement for it such as

if(block.nVersion > 3 || block.nVersion == 4)
For this case and lower cases rather than removing someone elses opportunity when forking

Reply:
`if (block.nVersion > 3 || block.nVersion == 4)` seems 'iffy, wouldn't this condition pass `true` *continously* above block 3? E.g: It would be true even at nVersion 5, 6, 7, etc... where nVersion 5 **removes** the element this condition is safeguarding (Condition should be `false` above or below block 4).

Also, if someone were to fork ZENZO's codebase, their first blocks would be nVersion 3 (No zerocoin / accumulators) and would become nVersion 4 after `height_start_ZC`, however... since ZC is dead, there is no reason for forking devs to keep this code, realistically they should start at nVersion 1 or 2, and zerocoin/accumulators should **never** be introduced in the first place (E.g: Zerocoin should be completely removed from forks), this code is only **needed** for ZENZO because we have previously used Zerocoin, in all fairness.

TL;DR: This code is only useful for ZENZO itself, due to previously using Zerocoin. Fresh forks should NOT keep Zerocoin in their codebase... therefore, I believe it's their responsibility to remove Zerocoin in the first place, it is an obsolete feature and only acts as baggage within our codebase.

Reply:
Correct in those instances but when it comes the time to bump to block version #5 you would add the condition similarly of

block.nVersion < 5 

Just keeping with how its seen being done by our upstream and such.

Just a suggestion of mine